Sources/Plasma/PubUtilLib/plPipeline/GL/plGLMaterialShaderRef.cpp

```diff
@@ -685,6 +685,16 @@ uint32_t plGLMaterialShaderRef::IHandleMaterial(uint32_t layer, std::shared_ptr<
     // Ignoring the bit about self-rendering cube maps
 
     plLayerInterface* currLay = /*IPushOverBaseLayer*/ fMaterial->GetLayer(layer);
+    
+    /*
+     If the layer opacity is 0, don't draw it. This prevents it from contributing to the Z buffer.
+     This can happen with some models like the fire marbles in the neighborhood that have some models
+     for physics only, and then can block other rendering in the Z buffer.
+     DX pipeline does this in ILoopOverLayers.
+     */
+    if(currLay->GetOpacity() <= 0) {
```

Comment:
Is there a risk of this impacting layer animations that start with 0 opacity and are intended to animate up to a non-0 value?

Reply:
Yes, I think I overlooked that. In the DX version, this check is also in IHandleMaterial, but it calls that every render pass. I'll move this check somewhere in the render pass.

Reply:
I updated this and moved the opacity check into the render loop. It's only checking the root layer of a pass, not sure if there could be additional issues if the base layer is transparent but another layer in the same pass isn't. I'm not sure how opacity is supposed to be inherited in the layer system.

The fragment function discarding could also be a fix for the issue. It would have the same effect, transparent fragments wouldn't contribute to the Z buffer. But it's more expensive to attempt to render a layer that is still completely invisible.

Reply:
That sound be fine, I'm pretty sure only the first layer in a pass sets the opacity for that pass.